packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -439,6 +440,35 @@ function WorklistDataGrid(props) {
 
   const handleMoveCellFocus = (fromCell, toCell) => {
     setFocusedRowCol(toCell.row, toCell.col, true);
+
+    // Obtain coordinate rectangles for grid container, column header, and new cell selection
+    const gridContainerRect = gridContainerRef.current.getBoundingClientRect();
+    const columnHeaderRect = grid.current.rows[0].cells[toCell.col].getBoundingClientRect();
+    const nextCellRect = grid.current.rows[toCell.row].cells[toCell.col].getBoundingClientRect();
+
+    // Calculate horizontal scroll offset for right boundary
+    if (nextCellRect.right > gridContainerRect.right) {
+      gridContainerRef.current.scrollBy(nextCellRect.right - gridContainerRect.right, 0);
+    } else {
+      // Calculate horizontal scroll offset for left boundary
+      let scrollOffsetX = 0;
+      if (pinnedColumnOffsets.length > 0) {
+        const lastPinnedColumnRect = grid.current.rows[toCell.row].cells[pinnedColumnOffsets.length - 1].getBoundingClientRect();
```

Comment:
Not really a comment but I'm wondering if at some point we could refactor this logic to more simplify specify the currently selected row and the newly selected row.